ledger/error/src/main/scala/com/daml/error/ErrorCategory.scala

```diff
@@ -0,0 +1,360 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.error
+
+import io.grpc.Status.Code
+import org.slf4j.event.Level
+
+import scala.concurrent.duration._
+
+/** Standard error categories
+  *
+  * Ideally, all products will return errors with appropriate error codes. Every such
+  * error code is associated with an error category that defines how the error will appear
+  * in the log file and on the api level.
+  */
+sealed trait ErrorCategory {
+
+  /** The Grpc code use to signal this error (in case it is signalled via API) */
+  def grpcCode: Option[Code]
+
+  /** The log level used to log this error on the server side */
+  def logLevel: Level
+
+  /** Default retryability information for this error category */
+  def retryable: Option[ErrorCategoryRetry]
+
+  /** If true, the event is security sensitive and error details should not be emitted on the api */
+  def securitySensitive: Boolean
+
+  /** Int representation of this error category */
+  def asInt: Int
+
+  /** Rank used to order severity (internal only) */
+  private[error] def rank: Int
+
+}
+
+object ErrorCategory {
+
+  val all: Seq[ErrorCategory] =
```

Comment:
As this is a constant this should be `All` (see [Scala Style Guide](https://docs.scala-lang.org/style/naming-conventions.html#constants-values-and-variables)).